### PR TITLE
fix: replace unwrap with Valid

### LINKED ISSUE #1739: fix: drop `unwrap` from `from_proto`
`from_proto` contains a few `unwrap`. Remove them and use `Valid` type here. Make sure to use `Valid::from_iter` for when one needs to check iterables.

### PR DESCRIPTION
**Pull request by emekaokoli19**
/claim #1739

**#1739**  
Fixes #1739  fix: drop unwrap from from_proto

**Build & Testing:**

- [ x] I ran `cargo test` successfully.
- [ x] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [x] I have added relevant unit & integration tests.
- [x] I have updated the [documentation] accordingly.
- [ x] I have performed a self-review of my code.
- [ x] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

## Summary by CodeRabbit

- **Refactor**
	- Improved error handling in data processing with the introduction of a new validation method.
	- Refined error handling and skipping logic in various parts of the code.

### CODE REVIEW

File: src/generator/from_proto.rs [hunk 1]
```diff
@@ -23,6 +23,36 @@ struct Context {
     query: String,
 }
 
+/// A custom enum representing either a valid value or an error.
+#[derive(Debug)]
+enum Valid<T> {
```
Comment (amitksingh1490) [inline src/generator/from_proto.rs:28]: actually valid is already defined in `valid.rs` we need to use that instead of defining a new one

**Comment by tusharmath** [comment]
Moving to draft until the comments are fixed. Feel free to mark it as read for review whenever you are done fixing the comments.

File: src/generator/from_proto.rs (hunk 1 above)
Comment (emekaokoli19) [reply src/generator/from_proto.rs]: Sorry for the delay, I have used the Valid in valid.rs

**Comment by tusharmath** [comment]
@emekaokoli19 Appreciate your contributions but unfortunately, this PR isn't going in the right direction. Feel free to take help from the contributor's channel on discord and reopen it whenever you are ready. 
Thanks once again!

**Pull request closed without merging**